my-app/src/app/modules/courses-page/courses-page.component.ts

```diff
@@ -0,0 +1,45 @@
+import {Component, ChangeDetectionStrategy, OnInit} from '@angular/core';
+
+import isEqual from "lodash/isEqual";
+
+import {COURSE_DATA} from "@assets/mocks/course-data.mock";
+
+import {Course} from "./core";
+
+import {FilterPipe} from "./core";
+
+@Component({
+  selector: 'app-courses-page',
+  templateUrl: './courses-page.component.html',
+  styleUrls: ['./courses-page.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [FilterPipe]
+})
+export class CoursesPageComponent implements OnInit {
+  public courses: Course[] | undefined;
+  public searchValue: string | undefined;
+
+  constructor(private filter: FilterPipe) {
+  }
+
+  ngOnInit(): void {
+    this.courses = COURSE_DATA;
```

Comment:
See, retrieving and manipulating data should be extracted into a service to prevent moving this across components.

Reply:
I divided task 5 into 2 MRs: task_5(preparation) and task_5 with new services, directives to avoid a huge MR. After merge this one I will add new MR